code/game/objects/items/toys.dm

```diff
@@ -1292,4 +1293,72 @@
 			user.put_in_hands(A)
 		else
 			new /obj/item/clothing/gloves/anchor_arms(get_turf(src.loc))
-		qdel(src)
\ No newline at end of file
+		qdel(src)
+
+/obj/item/toy/balloon/decoy
+	name = "inflatable decoy"
+	desc = "Use this to fool your enemies into thinking you're a balloon!"
+	icon_state = "decoy_balloon_deflated"
+	w_class = W_CLASS_TINY
+	col = null
+	inflated_type = /obj/item/toy/balloon/inflated/decoy
+	volume = 120	//liters
+	var/decoy_phrase = null
+
+/obj/item/toy/balloon/decoy/verb/record_phrase()
+	set name = "Record Decoy Phrase"
+	set category = "Object"
+	set src in usr
+
+	var/mob/M = usr
+	if(M.incapacitated())
+		return
+
+	var/N = input("Enter a stock phrase for your decoy to say:","[src]") as null|text
+	if(N)
+		decoy_phrase = N
+
+/obj/item/toy/balloon/decoy/inflate(mob/user, datum/gas_mixture/G)
+	var/obj/item/toy/balloon/inflated/decoy/D = ..()
+	if(!istype(D))
+		return
+	user.drop_item(D, force_drop = 1)
+	D.appearance = user.appearance
+	var/datum/log/L = new
+	user.examine(L)
+	D.desc = L.log
+	qdel(L)
+	if(decoy_phrase)
+		D.decoy_phrase = decoy_phrase
+
+/obj/item/toy/balloon/inflated/decoy
+	desc = "An inflated decoy balloon."
+	icon_state = "decoy_balloon_deflated"
+	w_class = W_CLASS_GIANT
+	density = 1
+	can_be_strung = FALSE
+	var/decoy_phrase = null
+	var/list/hit_sounds = list('sound/weapons/genhit1.ogg', 'sound/weapons/genhit2.ogg', 'sound/weapons/genhit3.ogg',\
+	'sound/weapons/punch1.ogg', 'sound/weapons/punch2.ogg', 'sound/weapons/punch3.ogg', 'sound/weapons/punch4.ogg')
+
+/obj/item/toy/balloon/inflated/decoy/examine(mob/user, var/size = "")
+	if(desc)
+		to_chat(user, desc)
+
+/obj/item/toy/balloon/inflated/attackby(obj/item/weapon/W, mob/user)
+	..()
+	if(!src.gcDestroyed)
```

Comment:
Being hit by an object causes the bopping, wobbling, and quote saying just like if you were to punch it.
However, I need that to not happen if the object hitting it causes it to pop. This means I need a way to check whether it's been popped after the `..()` call. Is there a better way to do that than checking `gcDestroyed`?